### DISCUSSION TITLE
Stuck on "SnappyMail" Loading Screen

### DISCUSSION THREAD

**Discussion by DawnWakening** [the-djmaze/snappymail#1113] [General]
Hi all, I'm hoping to get some guidance, I'm stuck! 

I've setup SnappyMail on my LiteSpeed server, running PHP 7.4, with all the extensions enabled. 

I have set the file and folder permissions, and confirmed the files/folders are owned by the correct user. 

Unfortunately, I'm not having much luck, as when I load the site, or /?admin panel, it's stuck on a page with a gray background with the text "SnappyMail". It simply doesn't load. 

I've enabled Debug Logging via SnappyMail, and unfortunately, there are no meaningful logs (Only INFO logs), no errors. 

I have also enabled full PHP logging, confirmed PHP logging works (making a test PHP script to cause a crash), and it appears there are no errors outputting from SnappyMail. What is happening? I've tried multiple versions of PHP, fresh installs, etc, but still absolutely no luck! 

I am using the latest .ZIP file (snappymail-2.27.3.zip), and can confirm all permissions are correct. 

I also moved the data folder outside of the main root directory, and can confirm all files are generated correctly once enabling the include.php file. I've got no idea, and I'm at wits end as to why it's not working...

**Comment by cm-schl** [comment]
Hi,

have you looked into the browser error console if maybe there is some error?
Does it happen with every browser?

**Reply by DawnWakening** [reply]
> Hi,
> 
> have you looked into the browser error console if maybe there is some error? Does it happen with every browser?

After having a look, it appears in Chrome, this error is thrown in the console: 
Uncaught SyntaxError: Invalid or unexpected token (at libs.min.js:1:1)
Uncaught SyntaxError: Invalid or unexpected token (at app.min.js:1:1)

In Firefox, I also get: 
Ruleset ignored due to bad selector. [app.min.css:178:192](https://xyz.xyz/snappymail/v/2.27.3/static/css/app.min.css)

When inspecting these specific files in the console, it appears their encoding isn't valid (see links below):
https://i.kutit.net/h/0XX9D2.png
https://i.kutit.net/h/EzA1Z1.png

Upon viewing the files in Notepad++ via WinSCP, the files seem to be valid and readable, they also appear to be encoded in UTF-8.

My PHP is also setup and its charset is set to UTF-8 as confirmed with an ini_get echo for default_charset. 

I have tried uploading the files manually, and also uploading the files via SSH. Same issue(s). 

I am running PHP 7.4 on CyberPanel, running OpenLiteSpeed web server. 

Doesn't matter what PHP version I run, I get the same Uncaught SyntaxErrors.

**Reply by DawnWakening** [reply]
> Hi,
> 
> have you looked into the browser error console if maybe there is some error? Does it happen with every browser?

I should also mention, yes, it happens in every browser. Indicating it could possibly be an error with the encoding of the file, or a server-level issue, however, I have ZERO issues running other PHP scripts with CSS and JS. So this has blown my mind as to why the server is outputting the JS and CSS files with the wrong charset?

**Reply by DawnWakening** [reply]
> Hi,
> 
> have you looked into the browser error console if maybe there is some error? Does it happen with every browser?

When loading the /?admin panel, it's still stuck on the SnappyMail page, however, there are no console errors. Err..

https://i.kutit.net/h/zP8ti1.png

**Comment by DawnWakening** [comment]
Although upon inspecting the Style editor > admin.min.css > it looks like the encoding is incorrect yet again? 

https://i.kutit.net/h/mykdq9.png
